docs/proposals/0000-support-interchangeable-build-and-run-implementations.md

```diff
@@ -0,0 +1,248 @@
+        
+  env:
+    - name: VCAP_APP_HOST
+      value: 0.0.0.0
+    - name: VCAP_APP_PORT
+      value: '8080'
+    - name: VCAP_SERVICES
+      valueFrom:
+        secretKeyRef:
+          key: VCAP_SERVICES
+          name: vcap-services-secret
+    - name: PORT
+      value: '8080'
+    - name: POD_NAME
+      valueFrom:
+        fieldRef:
+          apiVersion: v1
+          fieldPath: metadata.name
+    - name: CF_INSTANCE_GUID
+      valueFrom:
+        fieldRef:
+          apiVersion: v1
+          fieldPath: metadata.uid
+    - name: CF_INSTANCE_IP
+      valueFrom:
+        fieldRef:
+          apiVersion: v1
+          fieldPath: status.hostIP
+    - name: CF_INSTANCE_INTERNAL_IP
+      valueFrom:
+        fieldRef:
+          apiVersion: v1
+          fieldPath: status.podIP
+          
+  livenessProbe: #...
+  readinessProbe: #...
+  image: localregistry-docker-registry.default.svc.cluster.local:30050/cf-k8s-controllers/kpack/images/ff5b920f-c729-4284-9ae9-241d95b98528@sha256:7a7ed9b42905c6fabd71ea09eb22b97e0c638c9a341a70c84f9461e167c9f50f
+  imagePullSecrets:
+    - name: image-registry-credentials 
+  
+  instances: 1
+  
+  version: '0'
+```
+
+
+
+# Migration
+[migration]: #migration
+
+This will be a breaking change, but we haven't made any guarantee about upgrades. Existing installations will need to be
+torn down and any workloads will need to be pushed from scratch.
+
+There is one migration scenario we should consider: how can an operator migrate from one build/run reconciler to another
+on an existing cluster? We could mandate that each reconciler take possession of a Workload it's reconciled by setting
+an appropriate annotation on it. Reconcilers could then verify that a workload is either unowned or that it is the owner
+before working on it.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+- This adds complexity to our existing implementation, as a cost for the added flexibility.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- What other designs have been considered?
+    - Keeping the reconciler logic all inside of the `korifi` codebase, but with logic switches to handle different
+      workload implementations. However, this makes it difficult for third parties to implement their desired workflows,
+      and also makes it harder to experiment or deprecate behavior.
+
+- Why is this proposal the best?
+    - It makes it simple to extend Korifi with customized build by providing a clear, simple interface to external
+      reconcilers.
+
+- What is the impact of not doing this?
+    - Korifi will be difficult to extend with custom behavior.
+
+# Prior Art
+[prior-art]: #prior-art
+
+N/A
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Should we place these new resources in a new API group, and what should its name be?
+- Are `BuildWorkload` and `RunWorkload` good names? Do we want to add the `CF` prefix that all other resources currently
```

Comment:
In a past call we actually are thinking of removing the `CF` prefix from the other resources and adding convenience shortnames (e.g. `cfbuild`, `cfapp`, etc.) that we can use with `kubectl`. I can't think of anything better necessarily right now.